### PR TITLE
Feature/31797 testdb api

### PR DESCRIPTION
**Pull request by md-magenta**
Tilføjer et REST API til at resette databasen.

### CODE REVIEW

File: oio_rest/oio_rest/default-settings.toml [hunk 1]
```diff
@@ -98,3 +98,10 @@ path = ""
 [log]
 activity_log_path = "/var/log/mox/activity.log"
 trace_log_path = "/var/log/mox/trace.log"
+
+
+[testing_api]
+# If enabled, expose /testdb/* endpoint for setup, reset and teardown of a test
+# database. Useful for integration test from other componenets such as MO. Does
+# not work when running multi-threaded.
```
Comment (carlbordum) [inline oio_rest/oio_rest/default-settings.toml]: test => tests
componenets => components

Why does it not work when running multithreaded?

File: docker-compose.yml [hunk 2]
```diff
@@ -11,7 +11,12 @@ services:
     volumes:
       - .:/code
       - ./dev-environment/dev-settings.toml:/user-settings.toml
-    command: ["gunicorn", "--reload", "-b", "0.0.0.0:8080", "oio_rest:app"]
+    command: [
+    "gunicorn",
+    "--config", "docker/gunicorn-settings.py",
+    "--reload", # Reload on file changes
+    "--workers", "1", # Single threaded in dev
```
Comment (carlbordum) [inline docker-compose.yml]: Why?

File: oio_rest/oio_rest/default-settings.toml (hunk 1 above)
Comment (md-magenta) [reply oio_rest/oio_rest/default-settings.toml]: Se other reply.

File: docker-compose.yml (hunk 2 above)
Comment (md-magenta) [reply docker-compose.yml]: When MO hits several ``/testing/db-*` endpoints in a row, it alternates between the multiple workers and leaves idle db connections in the old threads.

**Pull request merged**